### PR TITLE
feat(risedev): support HDFS backend in risedev

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
HDFS guidance
https://www.notion.so/risingwave-labs/RisingWave-use-HDFS-Backend-66f02b2cd0494b1681c5cda4ee820adf?pvs=4

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: Makefile.toml [hunk 1]
```diff
@@ -38,9 +39,17 @@ if ${is_all_in_one_enabled}
 else
   set_env RISEDEV_CARGO_BUILD_CRATE "risingwave_cmd"
 end
+
+if ${is_hdfs_backend}
+  set_env BUILD_HDFS_BACKEND_CMD "-p risingwave_object_store --features hdfs-backend"
```
Comment (hzxa21) [inline Makefile.toml:44]: why do we need to add `-p risingwave_object_store`? I think adding `--features hdfs-backend` is good enough.

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

File: Makefile.toml (hunk 1 above)
Comment (wcy-fdu) [reply Makefile.toml:44]: It seems that `RISEDEV_CARGO_BUILD_CRATE` do not include `risingwave_object_store`. I will double check this and modify if necessary.

Comment (wcy-fdu) [reply Makefile.toml:44]: We need to add `-p risingwave_object_store`:
image

Comment (hzxa21) [reply Makefile.toml:44]: It includes `risingwave_cmd` or `risingwave_cmd_all`, which is basically the risingwave binary so `risingwave_object_store` should be transitively included.

Comment (wcy-fdu) [reply Makefile.toml:44]: `risingwave_cmd` do not have direct dependency of `rw_object_store`,, so let's keep it simple and do not add more dependency here.

**Pull request merged**